src/conv.rs

```diff
@@ -195,102 +195,75 @@ unsafe impl<'a, T: FromValue> FromValue for Option<T> {
 unsafe impl<'a, T: IntoValue> IntoValue for Option<T> {
     fn into_value(self, rt: &Runtime) -> Value {
         match self {
-            Some(y) => crate::frame!(rt: (x) {
-                x = y.into_value(rt);
-                unsafe { Value::some(rt, x) }
-            }),
+            Some(y) => unsafe { Value::some(rt, y) },
             None => Value::none(),
         }
     }
 }
 
 unsafe impl<'a> FromValue for &str {
     fn from_value(value: Value) -> Self {
-        let len = unsafe { crate::sys::caml_string_length(value.0) };
-        let ptr = unsafe { crate::sys::string_val(value.0) };
         unsafe {
-            let slice = ::core::slice::from_raw_parts(ptr, len);
-            ::core::str::from_utf8(slice).expect("Invalid UTF-8")
+            let len = crate::sys::caml_string_length(value.0);
+            let ptr = crate::sys::string_val(value.0);
+            let s = ::core::slice::from_raw_parts(ptr, len);
+            core::str::from_utf8_unchecked(s)
```

Comment:
FYI, in ocaml-interop I switched from `unchecked` to `lossy`, which adds a bit of overhead, but feels more correct for the default conversion: https://github.com/simplestaking/ocaml-interop/blob/798d46ac33e01eb5495f11d63174b5825fb2472c/src/conv/from_ocaml.rs#L64

Reply:
Oh, and another thing, creating slices from pointers to the OCaml heap is not very safe (because it is borrowing data from the OCaml heap that may get moved, or deallocated at any time without you knowing it), and less so in this case where there are no lifetimes involved. Is there any code that depends on `FromValue` being implemented for types like `&str` and `&[u8]` ?

Reply:
Nothing depends on them so it’s probably best to remove those.